### DISCUSSION TITLE
Deploys successfully but no controls react when clicked

### DISCUSSION THREAD

**Discussion by renlycat** [zauberzeug/nicegui#4063] [Q&A] [answered]
### Description

Hi,

I finally got my app deployed on IIS, and the page renders fine from the web server, but none of the controls are clickable.  Tab panels , switches, etc. are all dead clicks.  There is a select i'm populating that is showing values when clicked, but selecting one does nothing.  Any ideas?  Here is my main.py:

```py
def init(fastapi_app: FastAPI):
    ui.run_with(
        fastapi_app,
    )

fapp = FastAPI()
init(fapp)


def run_app():
    uvicorn.run("app.main:app", log_level="info")
```

web.config is defined as:

```
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
    <system.webServer>
        <handlers>
            <add name="Python FastAPI Handler"
                 path="*"
                 verb="*"
                 modules="httpPlatformHandler"
                 resourceType="Unspecified" />
        </handlers>
        <httpPlatform processPath="#{Octopus.Action.Package.InstallationDirectoryPath}\venv\Scripts\python.exe"
                      arguments="-m uvicorn app.main:fapp --host 0.0.0.0 --port %HTTP_PLATFORM_PORT%"
                      stdoutLogEnabled="true"
                      stdoutLogFile="%HOME%\LogFiles\python.log"
                      startupTimeLimit="60"
                      processesPerApplication="1">
            <environmentVariables>
                <environmentVariable name="SERVER_PORT" value="%HTTP_PLATFORM_PORT%" />
            </environmentVariables>
        </httpPlatform>
    </system.webServer>
</configuration>
```

I should note that it runs completely fine locally.  So close yet so far...

Thanks

**Comment by falkoschindler** [comment]
Does anyone from the community have experience with hosting FastAPI/NiceGUI apps on Microsoft IIS Web servers?

**Reply by renlycat** [reply]
Thanks, i should add that in the browser console i'm getting "reloading because connection timed out" when clicking on a control

**Comment by renlycat** [comment, marked as answer]
I was able to resolve this by enabling websockets in IIS, setting it as an allowed server variable and setting the value via the web.config:

https://www.oxygenxml.com/doc/versions/26.1.0/ug-waCustom/topics/WA-websocket.html
